### PR TITLE
Fix removing extra space from event

### PR DESCRIPTION
**Pull request by AnthonyGuerreiro**
If server sends the following event:
event:FooBar
data:{"foo":"bar baz"}

The first space after the first ':' is removed, becoming:
{"foo":"barbaz"}

This commit should first check if the second part of the line (after ':') starts with a white space.
If it doesn't then it won't replace any space in its content.

I assume this exists to remove the space for cases when the input is in the format:
data: {"foo":"bar baz"}

### CODE REVIEW

**Comment by jkodumal** [comment]
Thanks @AnthonyGuerreiro -- good catch. Would you mind adding a few test cases?

File: src/main/java/com/launchdarkly/eventsource/EventParser.java [hunk 1]
```diff
@@ -43,7 +43,10 @@ public void line(String line) {
       processComment(line.substring(1).trim());
     } else if ((colonIndex = line.indexOf(":")) != -1) {
       String field = line.substring(0, colonIndex);
-      String value = line.substring(colonIndex + 1).replaceFirst(" ", EMPTY_STRING);
+      String value = line.substring(colonIndex + 1);
+      if (value.charAt(0)==' ') {
```
Comment (jkodumal) [inline src/main/java/com/launchdarkly/eventsource/EventParser.java]: Note for posterity's sake: this is the correct behavior according to the spec:

https://www.w3.org/TR/eventsource/#parsing-an-event-stream

**Comment by AnthonyGuerreiro** [comment]
Actually, I introduced a bug with this fix. I fixed it already in my branch but you were quicker than me.
Not sure if I can somehow add an extra commit to this PR or need to do another one?

**Comment by jkodumal** [comment]
Feel free to open another PR.

**Comment by AnthonyGuerreiro** [comment]
Done (#26 ). Thank you!

**Pull request merged**